jax/typing.py

```diff
@@ -13,19 +13,54 @@
 # limitations under the License.
 
 """
+The JAX typing module is where JAX-specific static type annotations live.
 This submodule is a work in progress; to see the proposal behind the types exported
 here, see https://jax.readthedocs.io/en/latest/jep/12049-type-annotations.html.
 
-In short, for JAX-compatible functions, we recommend that input arrays be annotated with
-:class:`jax.typing.ArrayLike`, and outputs be annotated with :class:`jax.Array`.
-For example::
+The currently-available types are:
 
+- :class:`jax.Array`: annotation for any JAX array or tracer (i.e. representations of arrays
+  within JAX transforms).
+- :class:`jax.typing.ArrayLike`: annotation for any value that is safe to implicitly cast to
+  a JAX array; this includes :class:`jax.Array`, :class:`numpy.ndarray`, as well as Python
+  scalar values (e.g. :class:`int`, :class:`float`, etc.) and numpy scalar values
```

Comment:
Instead of "Python scalar", could say "Python builtin numeric"? AFAIK Python docs don't refer to those as "scalars"

Reply:
Done